Code/Framework/AzToolsFramework/AzToolsFramework/Entity/PrefabEditorEntityOwnershipService.cpp

```diff
@@ -167,14 +172,25 @@ namespace AzToolsFramework
 
     bool PrefabEditorEntityOwnershipService::DestroyEntity(AZ::Entity* entity)
     {
+        AZ_Assert(entity, "Tried to destroy a null entity");
         return DestroyEntityById(entity->GetId());
     }
 
     bool PrefabEditorEntityOwnershipService::DestroyEntityById(AZ::EntityId entityId)
     {
         AZ_Assert(IsInitialized(), "Tried to destroy an entity without initializing the Entity Ownership Service");
-        AZ_Assert(m_entitiesRemovedCallback, "Callback function for DestroyEntityById has not been set.");
         OnEntityRemoved(entityId);
+        if (auto owningInstance = m_instanceEntityMapperInterface->FindOwningInstance(entityId); owningInstance.has_value())
+        {
+            if (owningInstance->get().GetContainerEntityId() == entityId)
+            {
+                owningInstance->get().DestroyContainerEntity();
+            }
+            else
+            {
+                owningInstance->get().DestroyEntity(entityId);
+            }
+        }
         return true;
```

Comment:
looks like it always returns true (like what the function did before). Is it the right way of continuing do this? 🤔

Reply:
This should be fixed to return the correct outcome. I'll add that.